org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/panel/InfoPanel.java

```diff
@@ -2993,7 +2998,15 @@ public void sort(Comparator<Object> cmpr, boolean ascending) {
 		int col = lsc.getColumnIndex();
 		indexOrderColumn = col;
 		isColumnSortAscending = ascending;
-		sqlOrderColumn = p_layout[col].getColSQL().trim();
+		String displayColumn = p_layout[col].getDisplayColumn();
+		sqlOrderColumn = !Util.isEmpty(displayColumn) ? displayColumn : p_layout[col].getColSQL().trim();
+
+		// join tables to sort by display value
+		if(!Util.isEmpty(displayColumn) && (DisplayType.isLookup(p_layout[col].getAD_Reference_ID()) || DisplayType.isChosenMultipleSelection(p_layout[col].getAD_Reference_ID()))) {
+			MTable table = getTable(p_layout[col].getAD_Reference_Value_ID(), p_layout[col].getColumnName());
+			if(table != null && !table.getTableName().equals(joinTable))
+				joinTable = table.getTableName();
```

Comment:
perhaps better to rename this as joinTableForUserOrder

Reply:
variable renamed